system/HTTP/Header.php

```diff
@@ -28,15 +28,23 @@ class Header
     /**
      * The value of the header. May have more than one
      * value. If so, will be an array of strings.
+     * E.g.,
+     *   [
+     *       'foo',
+     *       [
+     *           'bar' => 'fizz',
+     *       ],
+     *       'baz' => 'buzz',
+     *   ]
      *
-     * @var array<array<string>|string>|string
+     * @var array<int|string, array<string, string>|string>|string
```

Comment:
This is indicative to me that we should probably be using a value object or collection class.

Reply:
I don't know why we need to have both `['bar' => 'fizz']` and `'bar' => 'fizz'`.

Reply:
Agreed! v5? 😂